network/handlers/drain_test.go

```diff
@@ -55,41 +109,56 @@ func TestDrainMechanics(t *testing.T) {
 		t.Errorf("Probe status = %d, wanted %d", got, want)
 	}
 
-	// Start to drain, and cancel the context when it returns.
+	// Start to drain, and close the channel when it returns.
 	done := make(chan struct{})
 	go func() {
 		defer close(done)
 		drainer.Drain()
 	}()
 
 	select {
-	case <-time.After(40 * time.Millisecond):
-		// Drain is blocking.
 	case <-done:
 		t.Error("Drain terminated prematurely.")
+	case <-init:
+		// OK.
 	}
+	mt.advance(40 * time.Millisecond)
```

Comment:
Given your fancy new mocking, why not 99 :)

Reply:
Why not even closer.